### PR TITLE
CM-32:Types of inheritance

### PR DESCRIPTION
**Pull request by flavia-tavani**
CM-32: Types Of Inheritance In C#

### CODE REVIEW

File: csharp-basic-topics/TypesOfInheritanceInCSharp/Tests/TypesOfInheritanceTest.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
+using System;
+using System.IO;
+using TypesOfInheritanceInCSharp;
+using Xunit;
+
+namespace TypesOfInheritanceInCSharpTests;
+public class TypesOfInheritanceTest
```
Comment (CodeMazeBlog) [inline csharp-basic-topics/TypesOfInheritanceInCSharp/Tests/TypesOfInheritanceTest.cs:7]: Newline

File: csharp-basic-topics/TypesOfInheritanceInCSharp/Tests/TypesOfInheritanceTest.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    [Fact]
+    public void WhenHaveSmartphone_ThenItOverridesDeviceCanMakePhoneCall()
+    {
+        var smartphone = new Smartphone();
+        smartphone.OperatingSystem = "iOS";
+        smartphone.Inches = 10;
+
+        Assert.True(smartphone.DeviceCanMakePhoneCall());
```
Comment (CodeMazeBlog) [inline csharp-basic-topics/TypesOfInheritanceInCSharp/Tests/TypesOfInheritanceTest.cs:56]: Add newlines between Arrange/Act/Assert wherever applicable.

File: csharp-basic-topics/TypesOfInheritanceInCSharp/TypesOfInheritanceInCSharp/Smartphone.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+public class Smartphone : MobileDevice
+{
+    public Smartphone() 
+    {
+        InstalledApps = new List<string>();
+    }
+
+    protected Smartphone(string operatingSystem)
+    {
+
```
Comment (CodeMazeBlog) [inline csharp-basic-topics/TypesOfInheritanceInCSharp/TypesOfInheritanceInCSharp/Smartphone.cs:12]: Remove newline

**Review by CodeMazeBlog — COMMENTED** [review]
The code is pretty clean, I've added some minor suggestions that should be addressed.

File: csharp-basic-topics/TypesOfInheritanceInCSharp/Tests/TypesOfInheritanceTest.cs (hunk 1 above)
Comment (flavia-tavani) [reply csharp-basic-topics/TypesOfInheritanceInCSharp/Tests/TypesOfInheritanceTest.cs:7]: It seems to me that this blank line is already added (line 5)

File: csharp-basic-topics/TypesOfInheritanceInCSharp/TypesOfInheritanceInCSharp/Smartphone.cs (hunk 3 above)
Comment (flavia-tavani) [reply csharp-basic-topics/TypesOfInheritanceInCSharp/TypesOfInheritanceInCSharp/Smartphone.cs]: I removed blank line

File: csharp-basic-topics/TypesOfInheritanceInCSharp/Tests/TypesOfInheritanceTest.cs (hunk 2 above)
Comment (flavia-tavani) [reply csharp-basic-topics/TypesOfInheritanceInCSharp/Tests/TypesOfInheritanceTest.cs:56]: I don't know if this means that I need to write var result = smartphone.DeviceCanMakePhoneCall(); and then Assert.True(result)

**Pull request merged**